nntrainer/tensor/tensor_pool.h

```diff
@@ -225,6 +225,120 @@ class TensorPool {
         spec.tensor->setData(pool[spec.token].tensor->getData());
   }
 
+  /**
+   * @brief request placeholder which will be not managed by this tensor pool
+   * but will be managed externally
+   *
+   * @param name Name of the tensor
+   * @param dim Tensor dimension
+   * @return Tensor* ptr to the tensor
+   *
+   * @note returns empty tensor which must be filled by the caller before use.
+   */
+  Tensor *placeholder(const std::string &name, const TensorDim &dim) {
+    return nullptr;
+  }
+
+  /**
+   * @brief     create a new tensor with the given spec.
+   *
+   * @param name Name of this tensor.
+   * @param dim Tensor dimension.
+   * @param exec_order The execution orders for this tensor.
+   * @param lifespan Lifespan of this tensor.
+   * @param init Initializer of the tensor.
+   *
+   * @return ptr to the created tensor
+   *
+   * @note returns empty tensor which will be filled when allocate is called.
+   * @note we assume that the caller checks if the exec_order and lifespan are
+   * compatible.
+   */
+  Tensor *create(const std::string &name, const TensorDim &dim,
```

Comment:
I highly recommend you to use the keyword `request` in function names as these tensors are not ready to use.
(`create` gives a notion that the tensor is created)

Reply:
manager will have `requestTensor()` which will be a wrapper. Plus I think this is actually a creation of a tensor which is just unallocated for the time being, and the user of tensor pool must know of this behavior before use.

I can change but to be clear, are you suggesting `create, view, placeholder, extend, createOrExtend, and reidentifySource` to be `requestCreate, requestView, requestPlaceholder, requestExtend, requestCreateOrExtend, and requestReidentifySource`?

Reply:
Confirmed offline that changing create -> request